### PR TITLE
search: retrieve loan history of a patron

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

File: invenio_circulation/search.py [hunk 1]
```diff
@@ -18,33 +18,48 @@ class LoansSearch(RecordsSearch):
...
             )
 
         for result in search.scan():
             if result.loanid:
                 yield result
+
+    @classmethod
+    def search_loans_by_patron_pid(cls, patron_pid=None):
+        """Retrieve loans of a patron."""
+        if patron_pid:
```
Comment (ntarocco) [inline invenio_circulation/search.py]: Maybe remove the if and the `None` as default value

File: invenio_circulation/search.py [hunk 2]
```diff
@@ -18,33 +18,48 @@ class LoansSearch(RecordsSearch):
...
+            search = cls()
+            search = search.filter("term", patron_pid=patron_pid)
+            for result in search.scan():
+                if result:
```
Comment (ntarocco) [inline invenio_circulation/search.py]: Can you double check if you can get rid of this `if result`?

**Review by ntarocco — APPROVED** [review]

File: invenio_circulation/search.py (hunk 2 above)
Comment (zzacharo) [reply invenio_circulation/search.py]: Yes indeed it returns empty list if there are no results, so I removed it!

File: invenio_circulation/search.py (hunk 1 above)
Comment (zzacharo) [reply invenio_circulation/search.py]: Done.

**Pull request merged**